lib/wm_util.ml

```diff
@@ -111,11 +111,71 @@ module MediaType = struct
     loop ranges
 end
 
-module Date = struct
-  open CalendarLib
+module type Date_sig = sig
+  type t
+  val now : unit -> t
+  val parse_rfc1123_date_exn : string -> t
+  val parse_rfc1123_date : string -> t option
+end
+
+module type PtimeNow_sig = sig
+  val now : unit -> Ptime.t
+end
+
+(** MakePtimeDate creates a Ptime based Date module, that can be used with the
+    Webmachine.Make(IO)(Date) functor.
+
+    Examples:
+    module DateMock = Wm_util.MakePtimeDate(struct
+      let now = fun () -> Ptime.v (17591, 0L)
+    end)
+
+    module DatePtimeClock = Wm_util.MakePtimeDate(struct
+      let now = Ptime_clock.now
+    end)
+
+    module DateMirage = Wm_util.MakePtimeDate(struct
+      let now = fun () -> Ptime.v @@ Pclock.now_d_ps clock
+    end)
+*)
+module MakePtimeDate(NOW:PtimeNow_sig) : Date_sig = struct
```

Comment:
I don't think this functor is necessary. since all it does is define a few functions that don't use the argument to the functor, and then it just re-exports the single functions that was provided in the argument. This suggests that the argument to `Webmachine.Make` should just be `PtimeNow_sig`, appropriately renamed.

Reply:
But that would force Ptime to be used. My current design would also allow to create webmachine without Ptime, but another time representation instead.

Reply:
Ah, because it's abstract. Maybe a simpler thing to do is to just fix the type to be an `int` which specifies the number of seconds since the UNIX epoch. Any further granularity isn't useful for this use-case anyway.